x-pack/plugins/alerting/server/alerts_service/alerts_service.ts

```diff
@@ -275,7 +275,7 @@ export class AlertsService implements IAlertsService {
     // check whether this context has been registered before
     if (this.registeredContexts.has(context)) {
       const registeredOptions = this.registeredContexts.get(context);
-      if (!isEqual(opts, registeredOptions)) {
+      if (!isEqual(omit(opts, 'shouldWrite'), omit(registeredOptions, 'shouldWrite'))) {
```

Comment:
The metric threshold alert registration options was actually used by multiple rule types within O11y so to limit the scope of this onboarding effort to a single rule type, I'm allowing different `shouldWrite` flags for the same context. Everything else should be the same (field maps, component template refs, etc).